scipy/signal/filter_design.py

```diff
@@ -4864,6 +4864,191 @@ def _design_notch_peak_filter(w0, Q, ftype, fs=2.0):
     return b, a
 
 
+def iircomb(w0, Q, N, ftype='notch', fs=2.0):
+    """
+    Design Nth-order IIR notching or peaking digital comb filter.
+
+    A notching comb filter is a band-stop filter with a narrow bandwidth
+    (high quality factor). It rejects a narrow frequency band and
+    leaves the rest of the spectrum little changed.
+
+    A peaking comb filter is a band-pass filter with a narrow bandwidth
+    (high quality factor). It rejects components outside a narrow
+    frequency band.
+
+    Parameters
+    ----------
+    w0 : float
+        Frequency to remove or pass from a signal. If `fs` is specified, this
+        is in the same units as `fs`. By default, it is a normalized scalar
+        that must satisfy  ``0 < w0 < 1``, with ``w0 = 1`` corresponding to
+        half of the sampling frequency.
+    Q : float
+        Quality factor. Dimensionless parameter that characterizes
+        notch filter -3 dB bandwidth ``bw`` relative to its center
+        frequency, ``Q = w0/bw``.
+    N : int
+        The order of the filter. Must be between 0 and 24.
+    ftype : {'notch', 'peak'}
+        The type of comb filter generated by the function. Default is 'notch'.
+    fs : float, optional
+        The sampling frequency of the signal. Default is 2.0.
+
+    Returns
+    -------
+    b, a : ndarray, ndarray
+        Numerator (``b``) and denominator (``a``) polynomials
+        of the IIR filter.
+
+    Raises
+    ------
+    ValueError
+        If `w0` is less than or equal to 0 or greater than or equal to
+        ``fs/2``, if `ftype` is not 'notch' or 'peak', if `N` is less than
+        or equal to 0 or greater than 24.
+
+    See Also
+    --------
+    iirnotch
+    iirpeak
+
+    References
+    ----------
+    .. [1] Sophocles J. Orfanidis, "Introduction To Signal Processing",
```

Comment:
There is no reference to `[1]_` above, might be worth adding a quick `Notes` section to say "For implementation details, see [1]_."

Also, typically for high-order IIR filters there can be issues with stability when represented in TF (b, a) form, hence why we have second-order-sections filtering. It might be worth mentioning that this is not the case for these filters, and why (a quick look suggests it's just because there is a single repeated pole, which won't suffer from precision loss like high-order filters with non-repeated poles will...?)

Reply:
Fixed.